platform-sdk/docs/proposals/consensus-layer/Consensus-Layer.md

```diff
@@ -0,0 +1,277 @@
+
+Each submodule will likewise be defined by a pair of JPMS modules -- an API module and an implementation module. By
+separating the API and implementation modules, we make it possible to supply multiple implementation modules (which is
+useful for various testing or future maintenance tasks), and we also support circular dependencies between modules.
+
+### Initialization of the Consensus Module
+
+When Execution starts, it will (at the appropriate time in its startup routine) create an instance of Consensus, and
+`initialize` it with appropriate arguments, which will be defined in detail in further documents. Critically,
+Consensus **does not maintain state**. Execution is wholly responsible for the definition and management of state. To
+start Consensus from a particular moment in time, Execution will need to initialize it with some information such as the
+witnesses of the round it wants to start from. It is by using this `initialize` method that Execution is able to create
+a Consensus instance that starts from genesis, or from a particular round.
+
+Likewise, if a node needs to reconnect, Execution will `destroy` the existing Consensus, and create a complete new one,
+and `initialize` it appropriately with information from the starting round, after having downloaded and initializing
+itself with the correct round. Reconnect therefore is the responsibility of Execution. Consensus does not have to
+consider reconnect at all.
+
+### Gossip
+
+The Gossip module is responsible for gossiping messages between peers. The actual gossip implementation is not described
+here, except to say, that it will be possible to define and implement both event-aware and event-agnostic gossip
+implementations either to a fully connected network or where the set of peers is a subset of the whole. Nor does this
+document dictate whether raw TCP, UDP, HTTP2, gRPC, or other network protocols are used. This will be left to the design
+documents for Gossip.
+
+![Gossip](gossip-system.png)
+
+Gossip is the only part of Consensus that communicates over the network with gossip peers. When Gossip is initialized,
+it is supplied a "roster". This roster contains the full set of nodes participating in gossip, along with their metadata
+such as RSA signing keys, IP addresses, and so forth. The Gossip module decides which peers to gossip with (using
+whatever algorithm it chooses).
+
+#### Events
+
+Gossip is event-oriented, meaning that it is given events to gossip, and emits events it receives through gossip. An
+implementation of Gossip could be based on a lower-level implementation based on bytes, but at the module level, it
+works in terms of events.
+
+When the Gossip module receives events through gossip, it *may* choose to perform some deduplication before sending
+them to Event Intake, but it is not required to do so.
+
+Honest nodes will **only** send valid events in *topological order* to peers. A peer may still receive events out of
```

Comment:
This is correct, all chatter variations allow out of order event delivery.

Reply:
It might be worth specifying that syn sends non-ancient events in topological ordering **with no gaps**. Technically speaking, a legal topological ordering may have gaps. In that case, the orphan buffer should delay the delivery of all events after that gap until the events in the gap are received or become ancient.

Reply:
I modified the wording so as to say that some gossip algorithms require topological ordering. The details of the gossip algorithms will be covered in the Gossip module's design spec (TBD). I just needed to motivate why we wait until topo ordering before sending to gossip, and the fact that some gossip algorithms require it, I think gives sufficient justification for just doing it all the time.